Tools/CMake/R.cmake

```diff
@@ -0,0 +1,743 @@
+#       done during the build stage and not configuration
+# - [ ] All the code inside the if(APPLE), and if(WIN32) should be turned into
+#       a CMake module. I leave this for later cleanup
+# - [ ] Both R package installer can be improved by some caching, now cleaning can be
+#       a bit brutal
+#
+
+list(APPEND CMAKE_MESSAGE_CONTEXT R)
+
+set(R_VERSION "4.1.3")
+set(R_VERSION_MAJOR_MINOR "4.1")
+set(CURRENT_R_VERSION ${R_VERSION_MAJOR_MINOR})
+
+if(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
+  set(R_DIR_NAME "${R_VERSION_MAJOR_MINOR}-arm64")
+else()
+  set(R_DIR_NAME "${R_VERSION_MAJOR_MINOR}")
+endif()
+
+if(WIN32)
+  if(CMAKE_SIZEOF_VOID_P EQUAL 8) # 64 bits
+    set(R_DIR_NAME "x64")
+  elseif(CMAKE_SIZEOF_VOID_P EQUAL 4) # 32 bits
+    set(R_DIR_NAME "i386")
+  endif()
+endif()
+
+# ------ Preparing REnv Paths
+#
+set(MODULES_SOURCE_PATH
+    ${PROJECT_SOURCE_DIR}/Modules
+    CACHE PATH "Location of JASP Modules")
+
+set(MODULES_BINARY_PATH
+    "${CMAKE_BINARY_DIR}/Modules"
+    CACHE PATH "Location of the renv libraries")
+set(MODULES_RENV_ROOT_PATH
+    "${MODULES_SOURCE_PATH}/renv-root"
+    CACHE PATH "Location of renv root directories")
+set(MODULES_RENV_CACHE_PATH
+    "${MODULES_BINARY_PATH}/renv-cache"
+    CACHE PATH "Location of renv cache directories")
+set(JASP_ENGINE_PATH
+    "${CMAKE_BINARY_DIR}/Desktop/"
+    CACHE PATH "Location of the JASPEngine")
+
+make_directory("${MODULES_BINARY_PATH}")
+make_directory("${MODULES_RENV_ROOT_PATH}")
+make_directory("${MODULES_RENV_CACHE_PATH}")
+
+# ------
+
+if(APPLE)
+
+  set(R_FRAMEWORK_PATH "${CMAKE_BINARY_DIR}/Frameworks")
+  set(R_HOME_PATH
+      "${R_FRAMEWORK_PATH}/R.framework/Versions/${R_DIR_NAME}/Resources")
+  set(R_LIBRARY_PATH "${R_HOME_PATH}/library")
+  set(R_OPT_PATH "${R_HOME_PATH}/opt")
+  set(R_EXECUTABLE "${R_HOME_PATH}/R")
+  set(R_INCLUDE_PATH "${R_HOME_PATH}/include")
+  set(RCPP_PATH "${R_LIBRARY_PATH}/Rcpp")
+  set(RINSIDE_PATH "${R_LIBRARY_PATH}/RInside")
+
+  cmake_print_variables(R_FRAMEWORK_PATH)
+  cmake_print_variables(R_HOME_PATH)
+  cmake_print_variables(R_LIBRARY_PATH)
+  cmake_print_variables(R_OPT_PATH)
+  cmake_print_variables(R_EXECUTABLE)
+  cmake_print_variables(RCPP_PATH)
+  cmake_print_variables(RINSIDE_PATH)
+
+  if(INSTALL_R_FRAMEWORK AND (NOT EXISTS
+                              ${CMAKE_BINARY_DIR}/Frameworks/R.framework))
+
+    if(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
+
+      set(R_PACKAGE_NAME "R-${R_VERSION}-${CMAKE_OSX_ARCHITECTURES}.pkg")
+      set(R_DOWNLOAD_URL
+          "https://cran.r-project.org/bin/macosx/big-sur-arm64/base/R-${R_VERSION}-arm64.pkg"
+      )
+      set(R_PACKAGE_HASH "4e702650f8967bc388ae31d897a4ae888dd6e89b")
+
+    else()
+
+      set(R_PACKAGE_NAME "R-${R_VERSION}.pkg")
+      set(R_DOWNLOAD_URL
+          "https://cran.r-project.org/bin/macosx/base/R-${R_VERSION}.pkg")
+      set(R_PACKAGE_HASH "45121f2c830b0cd7d180aee3fc4cd80d0de1e582")
```

Comment:
Should the hash maybe also be defined where `R_VERSION` is defined?
As in, now it looks like you would just need to update `R_VERSION` above and things work but they won't because the hashcheck will fail. Not a big problem though because you will notice soon enough

Reply:
Exactly, and I'm planning to redirect those paths to our internal binary repo in static, and make a map of different versions with their hash, etc.